math/ConstArithmetic.cpp

```diff
@@ -0,0 +1,193 @@
+    using Class = ConstArithmetic<T>;
+
+    ConstArithmetic(
+        OperatorFcn operatorFcn,
+        const T& constant,
+        size_t dimension
+    ): Pothos::Block(),
+       _constant(0),
+       _position(0),
+       _operator(operatorFcn)
+    {
+        const Pothos::DType dtype(typeid(T), dimension);
+        this->setupInput(0, dtype);
+        this->setupOutput(0, dtype);
+
+        this->registerCall(this, POTHOS_FCN_TUPLE(Class, constant));
+        this->registerCall(this, POTHOS_FCN_TUPLE(Class, setConstant));
+
+        this->registerCall(this, POTHOS_FCN_TUPLE(Class, constOperandPosition));
+        this->registerCall(this, POTHOS_FCN_TUPLE(Class, setConstOperandPosition));
+
+        this->registerProbe("constant");
+        this->registerSignal("constantChanged");
+
+        // Call setter to emit signal
+        this->setConstant(constant);
+    }
+
+    T constant() const
+    {
+        return _constant;
+    }
+
+    void setConstant(const T& constant)
+    {
+        _constant = constant;
+
+        this->updateCallable();
+        this->emitSignal("constantChanged", constant);
+    }
+
+    size_t constOperandPosition() const
+    {
+        return _position;
+    }
+
+    void setConstOperandPosition(size_t position)
+    {
+        if((0 == position) || (1 == position))
+        {
+            _position = position;
+            this->updateCallable();
+        }
+        else
+        {
+            throw Pothos::InvalidArgumentException(
+                      "Invalid position",
+                      std::to_string(position));
+        }
+    }
+
+    void work() override
+    {
+        const auto elems = this->workInfo().minElements;
+        if(0 == elems)
+        {
+            return;
+        }
+
+        auto* input = this->input(0);
+        auto* output = this->output(0);
+
+        const T* buffIn = input->buffer();
+        T* buffOut = output->buffer();
+
+        for(size_t elem = 0; elem < elems; ++elem)
+        {
+            buffOut[elem] = _operator.call(buffIn[elem]).template extract<T>();
+        }
+
+        input->consume(elems);
+        output->produce(elems);
+    }
+
+private:
+    T _constant;
+    size_t _position;
+    Pothos::Callable _operator;
+
+    void updateCallable()
+    {
+        _operator.unbind(0);
+        _operator.unbind(1);
+        _operator.bind(_constant, _position);
+    }
+};
+
+//
+// Registration
+//
+
+static Pothos::Block* makeConstArithmetic(
+    const Pothos::DType& dtype,
+    const std::string& operation,
+    const Pothos::Object& constant)
+{
+    #define ifTypeDeclareFactory__(type, opKey, func) \
+        if((Pothos::DType::fromDType(dtype, 1) == Pothos::DType(typeid(type))) && (opKey == operation)) \
+            return new ConstArithmetic<type>(func, constant.convert<type>(), dtype.dimension());
+    #define ifTypeDeclareFactory_(type) \
+        ifTypeDeclareFactory__(type, "ADD", [](const type& a, const type& b)->type {return a+b;}) \
```

Comment:
I think the bound operator stuff is going to be pretty slow. I would look at this block not as 4 operations 2 positions, but just 6 operations. Add, mult, div num, div denom, sub low, sub high. Pardon the names. The function pointers should just themselves include the for loop and operate over the array `const type& a, const type * b, size_t len`. And of course thats where you can swap out templated functions for simd functions should that ever be a thing.

Reply:
Done. "SUB" is now "X-K" and "K-X", etc.